servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAliveManager.java

```diff
@@ -240,6 +244,8 @@ void channelOutputShutdown() {
 
     void channelInputShutdown() {
         assert channel.eventLoop().inEventLoop();
+        cancelIfStateIsAFuture(inputShutdownFuture);
```

Comment:
do we need to keep `inputShutdownFuture state?
- if no: can we just always initiate shutdown when this method is called?
- if yes: can we we return early from this method if `inputShutdownFuture != null` (e.g. we are already shutting everything down)?

Reply:
the future is a timeout we are ready to wait for the input shutdown event. When we receive `ChannelInputShutdownReadComplete`, this method is invoked and we can cancel the future and finish the shutdown. 
if the timeout wins, we close the channel and I don't expect `ChannelInputShutdownReadComplete` to propagate. But even if it propagates, going though `channelHalfShutdown` once again should have no actual effect. wdyt?

I renamed `inputShutdownFuture` -> `inputShutdownTimeoutFuture`.

Reply:
makes sense (this method is notification that input has already changed, we don't want to leave the schedule task hanging). the event may propagate but as you say should be a noop.